frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -438,6 +470,169 @@ pub mod pallet {
             Ok(Some(T::WeightInfo::bond_and_stake()).into())
         }
 
+        // TODO: Update Readme.md!
+
+        /// Start unbonding process for the account Id - contract Id pair.
+        ///
+        /// Value being unbonded will be used in case staker has at least that amount staked.
+        /// If unstaking the specified `value` would take staker below the `T::MinimumStakingAmount`,
+        /// entire staked amount will be unstaked.
+        ///
+        /// If user attempts to unstake more than is available, everything will be unstaked.
```

Comment:
If we reject it, this means the API needs to be changed since we need to be able to unstake everything.
This also means UI update.

I prefer it this way since it makes more sense from the backend side. We need to ensure that
staked per contract doesn't fall below minimum staking amount.

For the front end part though, it can issue a warning to the user in case their unbond_and_unstake operation will take cause them to unstake everything.